docs/project-setup/filesystem-sync.md

```diff
@@ -0,0 +1,73 @@
+# Filesystem Sync with Unison
+
+The default NFS filesystems that Outrigger provides are very easy to setup and easy to share code and other files into
+your project containers, but NFS can be slow when writing, reading, or scanning thousands of files when compared to
+native filesystem performance. This can be a particular challenge when building your code base from a package manager onto
+a volume mounted NFS share.
+
+It also has the downside of not propagating filesystem modification events, which can be problematic if you wanted to have
+a process in a container monitor your files and run a process to rebuild or update when modification notifications are triggered.
+
+In order to better support these types of operations, Outrigger provides support for multi-directional filesystem syncing 
+(which includes filesystem notifications) called `unison`.  Your project root folder is sync'd via `unison` in to a sync 
+container and exposed as a Docker Volume to the rest of your application. This volume is then used as your mount point 
+instead of the NFS share and allows nearly native filesystem performance and features.
+
+## Sync Volume Name
+
+The name can be discovered in the following order
```

Comment:
The name of the sync volume can be set in a variety of ways. It is determined using the following precedence: